config/custom_dns.go

```diff
@@ -14,14 +14,56 @@ type CustomDNS struct {
 	RewriterConfig      `yaml:",inline"`
 	CustomTTL           Duration         `yaml:"customTTL" default:"1h"`
 	Mapping             CustomDNSMapping `yaml:"mapping"`
+	Zone                ZoneFileDNS      `yaml:"zone" default:""`
 	FilterUnmappedTypes bool             `yaml:"filterUnmappedTypes" default:"true"`
 }
 
 type (
 	CustomDNSMapping map[string]CustomDNSEntries
 	CustomDNSEntries []dns.RR
+
+	ZoneFileDNS struct {
+		RRs        CustomDNSMapping
+		configPath string
+	}
 )
 
+func (z *ZoneFileDNS) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var input string
+	if err := unmarshal(&input); err != nil {
+		return err
+	}
+
+	z.RRs = make(CustomDNSMapping)
+	result := z.RRs
```

Comment:
This doesn't matter in our code ATM, but can you change this so we do `result := make(CustomDNSMapping)` here, and only assign `z.RRs` at the very end?  
That way the method is a no-op in case of error, which can be useful if you're unmarshaling into an instance with existing data.